source/geometries/Next100Shielding.cc

```diff
@@ -343,123 +550,224 @@ namespace nexus {
+        }
+      }
+      else{ // lateral
+        if (G4UniformRand() < 0.5){
+          vertex = bubble_seal_lateral_gen_->GenerateVertex("INSIDE");
+          vertex.setX(vertex.x() + (pedestal_x_/2. + pedestal_lateral_beam_thickness + bubble_seal_thickness/2.));
+        }
+        else{
+          vertex = bubble_seal_lateral_gen_->GenerateVertex("INSIDE");
+          vertex.setX(vertex.x() - (pedestal_x_/2. + pedestal_lateral_beam_thickness + bubble_seal_thickness/2.));
+        }
       }
-
-      else{    //LATERAL BEAM STRUCTURE
-	G4double rand_beam = int (8 * G4UniformRand());
-	// std::cout<< "viga numero "<<rand_beam<<std::endl; //0-7
-	if (rand_beam == 0) {
-	  //lat_1 (lat_beam_x,-beam_base_thickness_/2.,lateral_z_separation_/2.)
-	  vertex = lat_beam_gen_->GenerateVertex("INSIDE");
-	}
-	else if (rand_beam ==1){
-	  // //lat_2 (lat_beam_x,-beam_base_thickness_/2.,-lateral_z_separation_/2.)
-	  vertex = lat_beam_gen_->GenerateVertex("INSIDE");
-	  vertex.setZ(vertex.z() -lateral_z_separation_);
-	}
-	else if (rand_beam ==2){
-	  // //lat_3 	(-lat_beam_x,-beam_base_thickness_/2.,lateral_z_separation_/2.)
-	  vertex = lat_beam_gen_->GenerateVertex("INSIDE");
-	  vertex.setX(vertex.x() -(shield_x_ + 2*steel_thickness_ + lead_thickness_ ));
-	}
-	else if (rand_beam ==3){
-	  // //lat_4 (-lat_beam_x,-beam_base_thickness_/2.,-lateral_z_separation_/2.)
-	  vertex = lat_beam_gen_->GenerateVertex("INSIDE");
-	  vertex.setX(vertex.x() -(shield_x_ + 2*steel_thickness_ + lead_thickness_ ));
-	  vertex.setZ(vertex.z() -lateral_z_separation_);
-	}
-	else if (rand_beam ==4){
-	  // //lat_5 front_beam (-front_x_separation_/2.,-beam_base_thickness_/2.,front_beam_z)
-	  vertex = front_beam_gen_->GenerateVertex("INSIDE");
-	}
-	else if (rand_beam ==5){
-	  // //lat_6 front_beam (front_x_separation_/2.,-beam_base_thickness_/2.,front_beam_z)
-	  vertex = front_beam_gen_->GenerateVertex("INSIDE");
-	  vertex.setX(vertex.x() + front_x_separation_);
-	}
-	else if (rand_beam ==6){
-	  // //lat_7 front_beam (-front_x_separation_/2.,-beam_base_thickness_/2.,-front_beam_z)
-	  vertex = front_beam_gen_->GenerateVertex("INSIDE");
-	  vertex.setZ(vertex.z() -(shield_z_+2*steel_thickness_+lead_thickness_));
-	}
-	else if (rand_beam ==7){
-	  //lat_8 front_beam (front_x_separation_/2.,-beam_base_thickness_/2.,-front_beam_z)
-	  vertex = front_beam_gen_->GenerateVertex("INSIDE");
-	  vertex.setX(vertex.x() + front_x_separation_);
-	  vertex.setZ(vertex.z() -(shield_z_+2*steel_thickness_+lead_thickness_));
-	}
       }
 
+    else if(region=="EDPM_SEAL"){
+      G4double rand = G4UniformRand();
+      if (rand<perc_edpm_front_vol_){ // front
+        if (G4UniformRand() < 0.5){
+          vertex = edpm_seal_front_gen_->GenerateVertex("INSIDE");
+          vertex.setZ(vertex.z() + (shield_z_/2. + steel_thickness_ + edpm_seal_thickness/2.));
```

Comment:
The vertices are being generated into the `LEAD_BOX` volume, but this is not the intended behaviour, right? They should be generated just on the surface of the steel box, so either in the `SHIELDING_STEEL` or in the `INNER_AIR` volume. I think the displacement should be 
`vertex.setZ(vertex.z() + (shield_z_/2. - edpm_seal_thickness/2.));`, do you agree?

Reply:
It is not still clear, since from the slides it seems that the steel is above the seal. Let's wait for Jordi's answer to finally set the exact position.

Reply:
After Jordi's confirmation, we move the EDPM vertex to be generated from inside the INNER_AIR.